template-compose/app/src/main/java/co/nimblehq/template/compose/di/modules/DataStoreModule.kt

```diff
@@ -0,0 +1,27 @@
+package co.nimblehq.template.compose.di.modules
+
+import android.content.Context
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.*
+import androidx.datastore.preferences.preferencesDataStoreFile
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.android.qualifiers.ApplicationContext
+import dagger.hilt.components.SingletonComponent
+import javax.inject.Singleton
+
+private const val USER_PREFERENCES = "user_preferences"
```

Comment:
Shall we ty to keep the naming for template code as general as possible?

- `USER_PREFERENCES` -> `PREFERENCES`
- `UserPreferencesRepository` -> `PreferencesRepository`
- `UserPreferences` -> `Preferences`

Reply:
@Tuubz Good idea. 👍🏻 I have no idea about this. 😂 Will update.